### ISSUE TITLE
`estimate_memory_usage` throws `TypeError` when using HPO for `GBM` model in `TabularPredictor`

### ISSUE THREAD

**Issue by bjakubowskiVNSNY** [autogluon/autogluon#3928]
## Description 

- This line is throwing a `TypeError` when HPO is used to tune a `GBM` fit via `TabularPredictor`

https://github.com/autogluon/autogluon/blob/e4b7315c37e83adab537235fe9ef8b540eac8442/tabular/src/autogluon/tabular/models/lgb/lgb_model.py#L86

### Reproducible test case

```python
from autogluon.common.space import Int
from autogluon.tabular import TabularDataset, TabularPredictor

data_url = 'https://raw.githubusercontent.com/mli/ag-docs/main/knot_theory/'
train_data = TabularDataset(f'{data_url}train.csv')
label = 'signature'

predictor = TabularPredictor(label=label).fit(
    train_data.head(1000),
    hyperparameters={
        'GBM': {
            'num_leaves': Int(8,64,32),
            'ag_args':{'hyperparameter_tune_kwargs':'auto'}
        }
    }
)
```

Here's the stack trace and error

```
Warning: Exception caused LightGBM to fail during hyperparameter tuning... Skipping this model.
Traceback (most recent call last):
  File "/opt/conda/lib/python3.10/site-packages/autogluon/core/trainer/abstract_trainer.py", line 2150, in _train_single_full
    hpo_models, hpo_results = model.hyperparameter_tune(
  File "/opt/conda/lib/python3.10/site-packages/autogluon/core/models/abstract/abstract_model.py", line 1469, in hyperparameter_tune
    self._validate_fit_memory_usage(**kwargs)
  File "/opt/conda/lib/python3.10/site-packages/autogluon/core/models/abstract/abstract_model.py", line 1746, in _validate_fit_memory_usage
    approx_mem_size_req = self.estimate_memory_usage(**kwargs)
  File "/opt/conda/lib/python3.10/site-packages/autogluon/core/models/abstract/abstract_model.py", line 1644, in estimate_memory_usage
    return self._estimate_memory_usage(**kwargs)
  File "/opt/conda/lib/python3.10/site-packages/autogluon/tabular/models/lgb/lgb_model.py", line 86, in _estimate_memory_usage
    histogram_mem_usage_bytes = 20 * max_bins * len(X.columns) * num_leaves
TypeError: unsupported operand type(s) for *: 'int' and 'Int'
unsupported operand type(s) for *: 'int' and 'Int'
```

#### Version of AutoGluon being used, the version of pytorch

This MRE uses:
- `Python 3.10.13 (main, Oct  5 2023, 22:56:59) [GCC 9.4.0] on linux`
- `torch==2.0.1`
- `autogluon==1.0.0`

I haven't made modifications relevant to the bug, and there's nothing unusual about my environment or deployment

**Comment by Innixma** [comment]
Thanks for the bug report! I've reproduced the issue and sent a PR with a fix: #3931

**Comment by bjakubowskiVNSNY** [comment]
Awesome thanks much @Innixma will look forward to 1.0.1 release
